libraries/lib-stretching-sequence/AudioSegment.h

```diff
@@ -0,0 +1,22 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSegment.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+
+#pragma once
+
+#include "SampleCount.h"
+
+class AudioSegment
+{
+public:
+   virtual ~AudioSegment() = default;
+   virtual size_t GetFloats(float* const* buffers, size_t numSamples) = 0;
```

Comment:
This interface looks strange -- I don't see what tells it how many `buffers` are there.

Reply:
The effective number of channels is known in the context instances are used by both implementation and user. But you're right, looking back that's unusual. I'll pass a vector reference instead.